opencti-platform/opencti-graphql/src/utils/stix-filtering/stix-testers.ts

```diff
@@ -0,0 +1,378 @@
+  }
+  return false;
+};
+
+/**
+ * RELATION FROM
+ * - depending on stix type (relation or sighting), we might search in target_ref or where_sighted_refs (plurals!)
+ */
+export const testRelationTo = (stix: any, filter: Filter) => {
+  if (stix.type === STIX_TYPE_RELATION) {
+    const stixValues: string[] = toValidArray(stix.target_ref);
+    return testStringFilter(filter, stixValues);
+  }
+  if (stix.type === STIX_TYPE_SIGHTING) {
+    const stixValues: string[] = stix.where_sighted_refs ?? [];
+    return testStringFilter(filter, stixValues);
+  }
+  return false;
+};
+
+/**
+ * RELATION FROM TYPES
+ * - depending on stix type (relation or sighting), we might search in source_type or sighting_of_type (in extension)
+ * - we must also search in parent types
+ */
+export const testRelationFromTypes = (stix: any, filter: Filter) => {
+  if (stix.type === STIX_TYPE_RELATION) {
+    const stixValue = stix.extensions?.[STIX_EXT_OCTI].source_type;
+    const extendedStixValues: string[] = [...toValidArray(stixValue), ...getParentTypes(stixValue)];
+    return testStringFilter(filter, extendedStixValues);
+  }
+  if (stix.type === STIX_TYPE_SIGHTING) {
+    const stixValue = stix.extensions?.[STIX_EXT_OCTI].sighting_of_type;
+    const extendedStixValues: string[] = [...toValidArray(stixValue), ...getParentTypes(stixValue)];
+    return testStringFilter(filter, extendedStixValues);
+  }
+  return false;
+};
+
+/**
+ * RELATION TO TYPES
+ * - depending on stix type (relation or sighting), we might search in target_type or where_sighted_types (in extension)
+ * - we must also search in parent types
+ */
+export const testRelationToTypes = (stix: any, filter: Filter) => {
+  if (stix.type === STIX_TYPE_RELATION) {
+    const stixValue: string = stix.extensions?.[STIX_EXT_OCTI].target_type;
+    const extendedStixValues: string[] = [...toValidArray(stixValue), ...getParentTypes(stixValue)];
+    return testStringFilter(filter, extendedStixValues);
+  }
+  if (stix.type === STIX_TYPE_SIGHTING) {
+    const stixValues: string[] = stix.extensions?.[STIX_EXT_OCTI].where_sighted_types ?? [];
+    const extendedStixValues = [...stixValues, ...stixValues.map((t) => getParentTypes(t)).flat()];
+    return testStringFilter(filter, extendedStixValues);
+  }
+  return false;
+};
+
+/**
+ * REFS
+ * - uses a refs extractor using the standardId generator
+ */
+export const testRefs = (stix: any, filter: Filter) => {
+  const stixValues: string[] = stixRefsExtractor(stix, generateStandardId);
+  return testStringFilter(filter, stixValues);
+};
+
+/**
+ * INSTANCE TYPE (elementId)
+ * - elementId is id in stix extension
+ * - useSideEventMatching arg to optionally test against relations and refs of the object
+ */
+export const testInstanceType = (stix: any, filter: Filter, useSideEventMatching = false) => {
+  const stixValues: string[] = toValidArray(stix.extensions?.[STIX_EXT_OCTI]?.id);
+  if (!useSideEventMatching) {
```

Comment:
The case `useSideEventMatching` will be handled by a new filter key as we do not want exotic behavior like this (we try to have all filtering techniques embedded as filter keys.

Reply:
new filter key `connectedToId` encapsulate this case now